client/me/purchases/confirm-cancel-purchase/load-endpoint-form.js

```diff
@@ -92,35 +92,33 @@ function getFormData( { form, selectedPurchase, selectedSite } ) {
 function initializeDomainCancelForm( options ) {
 	const { form } = options,
 		domainCancelReason = form.querySelector( '#domain_cancel_reason' ),
-		confirmCheckbox = form.querySelector( '#confirm' ),
-		submitButton = form.querySelector( 'input[type=submit]' );
+		reasonsDiv = form.querySelector( '#domain_cancel_reasons' );
 
-	submitButton.disabled = ! form.querySelector( '#confirm' ).checked;
+	domainCancelReason.addEventListener( 'change', ( event ) => {
+		showDomainCancelReasonDetail( reasonsDiv, event.target.value );
+	} );
 
-	form.addEventListener( 'submit', ( event ) => {
-		event.preventDefault();
+	toArray( reasonsDiv.children ).forEach( ( div ) => {
+		const confirmCheckbox = div.querySelector( 'input[type=checkbox]' ),
+			submitButton = div.querySelector( 'input[type=submit]' );
 
-		if ( ! form.querySelector( '#confirm' ).checked ) {
-			return;
-		}
+		if ( confirmCheckbox && submitButton ) {
+			submitButton.disabled = ! confirmCheckbox.checked;
 
-		submitForm( options );
+			confirmCheckbox.addEventListener( 'click', ( event ) => {
+				submitButton.disabled = ! event.target.checked;
+			} );
+		}
 	} );
 
-	domainCancelReason.addEventListener( 'change', ( event ) => {
-		showDomainReasonDetail( {
-			form,
-			selectValue: event.target.value
-		} );
-	} );
+	form.addEventListener( 'submit', ( event ) => {
+		event.preventDefault();
```

Comment:
@fabianapsimoes do you think we should implement it in v1? It sounds like a nice UX improvement. In v2 we should rather replace this whole `Confirm Cancellation` page with rewritten version based on design team prototype.